db-functions

```diff
@@ -282,7 +282,7 @@ getpkgfile() {
 
 getpkgfiles() {
 	local f files
-	if [[ ! -z "$(printf '%s\n' "${@%\.*}" | sort | uniq -D)" ]]; then
+	if ! printf '%s\n' "${@%\.*}" | awk 'a[$0]++>1{exit 1}'; then
```

Comment:
Why not fix this all at once instead of changing the same line multiple times?

Reply:
It's two logically distinct changes.

Reply:
And so are several other changes you lumped into the first commit.

Reply:
They're all "this array wasn't quoted, but we can do a better fix than simply quoting it"